src/main/java/com/iota/iri/crypto/PearlDiver.java

```diff
@@ -22,22 +30,15 @@
     private volatile State state;
     private final Object syncObj = new Object();
 
-    public void cancel() {
-        synchronized (syncObj) {
-            state = State.CANCELLED;
-        }
-    }
-
-    private static void validateParameters(byte[] transactionTrits, int minWeightMagnitude) {
-        if (transactionTrits.length != TRANSACTION_LENGTH) {
-            throw new RuntimeException(
-                "Invalid transaction trits length: " + transactionTrits.length);
-        }
-        if (minWeightMagnitude < 0 || minWeightMagnitude > CURL_HASH_LENGTH) {
-            throw new RuntimeException("Invalid min weight magnitude: " + minWeightMagnitude);
-        }
-    }
-
+    /**
+     * Searches for a nonce such that the hash ends with {@code minWeightMagnitude} zeros.<br>
```

Comment:
To add the` {@value TransactionViewModel#NOUNCE_TRINARY_SIZE}` trits long nounce `{@code transactionTrits}` are changed from the following offset: `{@value TransactionViewModel#NOUNCE_TRINARY_OFFSET}`